net/batman-adv-legacy/files/lib/netifd/proto/batadv-legacy.sh

```diff
@@ -30,4 +30,4 @@ proto_batadv_teardown() {
 	(echo "none" > "/sys/class/net/$iface/batman_adv/mesh_iface") 2>/dev/null
 }
 
-add_protocol batadv
+add_protocol batadv-legacy
```

Comment:
I don't think this would work, as the function names above don't match. Replace `batadv` with `batadv_legacy` in the function names and adjust the add_protocol (and for consistency also the file name).